packages/svelte/src/reactivity/map.js

```diff
@@ -56,13 +56,17 @@ export class SvelteMap extends Map {
 	#sources = new Map();
 	#version = state(0);
 	#size = state(0);
+	/**@type {Reaction | null} */
+	#initial_reaction = null;
```

Comment:
Wondering if there are cases when map still lives, while the reaction could have been GCed and this would prevent it?

Reply:
Yeah I was wondering about that too...technically it shouldn't be the case because either there's no active reaction (it's outside of a derived/effect) or it's inside it but this means that when the derived/effect is no longer used the function should be GCd and everything inside it too...but I need to do a better check

Reply:
Maybe this can become `initial_reaction = WeakSet(active_reaction)` and we can check with `this.initial_reaction.has(active_reaction)`?